WalletWasabi.Fluent/Behaviors/ShowAttachedFlyoutWhenFocusedBehavior.cs

```diff
@@ -0,0 +1,126 @@
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Diagnostics;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.VisualTree;
+using Avalonia.Xaml.Interactivity;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Fluent.Helpers;
+
+namespace WalletWasabi.Fluent.Behaviors;
+
+public class ShowAttachedFlyoutWhenFocusedBehavior : Behavior<Control>
+{
+	public static readonly StyledProperty<bool> IsFlyoutOpenProperty =
+		AvaloniaProperty.Register<ShowAttachedFlyoutWhenFocusedBehavior, bool>(
+			nameof(IsFlyoutOpen));
+
+	private readonly CompositeDisposable _disposables = new();
+
+	private FlyoutController? _flyoutController;
+
+	public bool IsFlyoutOpen
+	{
+		get => GetValue(IsFlyoutOpenProperty);
+		set => SetValue(IsFlyoutOpenProperty, value);
+	}
+
+	protected override void OnAttachedToVisualTree()
+	{
+		base.OnAttachedToVisualTree();
+
+		if (AssociatedObject is null)
+		{
+			return;
+		}
+
+		var flyoutBase = FlyoutBase.GetAttachedFlyout(AssociatedObject);
+		if (flyoutBase is null)
+		{
+			return;
+		}
+
+		if (AssociatedObject.GetVisualRoot() is not Control visualRoot)
+		{
+			return;
+		}
+
+		_flyoutController = new FlyoutController(flyoutBase, AssociatedObject)
+			.DisposeWith(_disposables);
+
+		DescendantPressed(visualRoot)
+			.Select(descendant => AssociatedObject.IsVisualAncestorOf(descendant))
+			.Do(isAncestor =>
+			{
+				_flyoutController.IsOpen = isAncestor;
+				IsFlyoutOpen = isAncestor;
+			})
+			.Subscribe()
+			.DisposeWith(_disposables);
+
+		Observable.FromEventPattern(AssociatedObject, nameof(InputElement.GotFocus))
+			.Do(_ =>
+			{
+				_flyoutController.IsOpen = true;
+				IsFlyoutOpen = true;
+			})
+			.Subscribe()
+			.DisposeWith(_disposables);
+
+		Observable.FromEventPattern(AssociatedObject, nameof(InputElement.LostFocus))
+			.Where(_ => !IsFocusInside(flyoutBase))
+			.Do(_ =>
+			{
+				_flyoutController.IsOpen = false;
+				IsFlyoutOpen = false;
+			})
+			.Subscribe()
+			.DisposeWith(_disposables);
+
+		this.GetObservable(IsFlyoutOpenProperty).Subscribe(b => _flyoutController.IsOpen = b);
+
+		if (IsFlyoutOpen)
+		{
+			_flyoutController.IsOpen = false;
+		}
```

Comment:
If the Flyout is open then close it? Is this a workaround?

Reply:
Yes, this is a workaround for the case when the user switches theme. 
The same behavior is detached and re-attached on theme changes.
If you don't close it, the Flyout will show in an incorrect position. Theme changing is tricky!